javascript/Spread_operator.md

```diff
@@ -0,0 +1,301 @@
+
+#### Add to the Beginning of an Array Using unshift()
+The unshift() method will add an element to the beginning of an array, while its twin function, shift(), will remove one element from the beginning of the array.
+
+To add an element to the beginning of an array using unshift() try this:
+```
+var list = ["foo", "bar"];
+list.unshift("baz");
+["baz", "foo", "bar"] // result
+```
+To add multiple elements to the beginning of an array using unshift() try this:
+```
+var list = ["foo", "bar"];
+list.unshift("baz", "qux");
+["baz", "qux", "foo", "bar"] // result
+```
+If you need to add elements to the beginning of an array, the unshift() method will almost always be your simplest and quickest option.
+
+#### Add Element(s) to the Middle of an Array Using splice()
+The splice() method modifies the content of an array by removing existing elements and/or adding new elements.
+
+Below you’ll find the proper splice() function syntax:
+```
+array.splice( start, deleteCount [, item1 [, item2 [, ...] ] ] )
+```
+If you want to insert an element (or elements) into a particular point somewhere within the array, besides the beginning or end, then you should most likely be using the splice() method.
+
+To use splice() your code should look like this:
+```
+var list = ["foo", "bar"];
+list.splice( 1, 0, "baz"); // at index position 1, remove 0 elements, then add "baz" to that position
+                           // element "bar" will now automatically be moved to index position 2
+["foo", "baz", "bar"] // result
+```
+To add multiple elements in the middle of an array using splice() try this:
+
+```
+var list = ["foo", "bar"];
+list.splice( 1, 0, "baz", "qux");
+["foo", "baz", "qux", "bar"] // result
+```
+The splice command is like the Swiss Army Knife of array manipulation; however, you should first try using the much simpler push or unshift commands before using splice() to add to an array.
+
+#### Using Splice to Remove Array Elements in JavaScript
+The splice method can be used to add or remove elements from an array. The first argument specifies the location at which to begin adding or removing elements. The second argument specifies the number of elements to remove. The third and subsequent arguments are optional; they specify elements to be added to the array.
+
+Here we use the splice method to remove two elements starting from position three (zero based index):
+```
+var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];var removed = arr.splice(2,2);/*removed === [3, 4]arr === [1, 2, 5, 6, 7, 8, 9, 0]*/
+```
+An array containing the removed elements is returned by the splice method. You can see the removed array contains [3, 4] and the original array contains the remaining values.
+
+The splice method can also be used to remove a range of elements from an array.
+```
+["bar", "baz", "foo", "qux"]list.splice(0, 2) // Starting at index position 0, remove two elements ["bar", "baz"] and retains ["foo", "qux"].
+```
+
+
```

Comment:
@alodahl what is the need of these topics here? Because this tutorial is on Spread operator not on array methods.

Reply:
Oh gosh, you are right.  Some of them do use the spread operator in the syntax, but @urbashi123 if the point doesn't involve spreading somehow, or adding it doesn't make sense, please remove that part and adjust the Projected Times.

Reply:
line 73 and 74 are a good example of how spread can be included.

Reply:
So which part to keep from line 116 to 189? @Aayush-1999

Reply:
Keep only those examples which are related to spread operator or which shows the use of spread operator in one way or the other.